uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/request-ctrl.js

```diff
@@ -934,6 +942,7 @@ define([
 
         // Init the `balance` object based on the first absence type
         this.balance.opening = this.selectedAbsenceType.remainder;
+        this.calculateClosingBalanceChange();
```

Comment:
Also why are we calculating opening or closing balance inside `initAbsenceType` function. it does not make sense. 
After looking at the code it seems that we are calculating because when `changeInNoOfDays` is called, we also need to reassign balance values. So instead of only calculating that, we are calling `initAbsenceType` function which again does not make sense. 

So 
1. `initAbsenceType` call should be removed from `changeInNoOfDays` and only the code to recalculate should be added inside.
2. `this.balance.opening = this.selectedAbsenceType.remainder;` should be removed from `initAbsenceType` and the calculation for both opening and closing balance should go to the new function which you created `calculateOpeningAndClosingBalance`
3. `calculateOpeningAndClosingBalance` should be called from `initAfterContactSelection`, `changeInNoOfDays` and `calculateBalanceChange`

I know all of this is not part of the ticket, but if we find some code which can be improved ,we always should refactor it.